### PR TITLE
Add magazines to 223 rifles [RDY]

### PR DESCRIPTION
**Pull request by mugling**
This PR follows #14981 and adds magazines to all 223 rifles without an integral magazine.

Presently they spawn without magazines requiring you to first craft a heavy and unreliable _survivor 223 magazine_ which is also very slow to reload. Better magazine types are not craftable and must be be scavenged. The most plentiful is the the 10-round _Ruger 223 magazine_ which fits almost all 223 rifles.

The 30-round _STANAG magazine_ is rarer and the bulky _STANAG drum_ is only found at military sites. The latter is too large to fit into a standard ammo pouch (#14972). You can craft wearable ammo belts (separate PR) which have weight and volume proportional to their length. These are however less reliable than drum magazines.

The leadworks guns take unique and incompatible magazines which are made of a lightweight polymer and have higher capacities. This is offset by their scarcity and the lack of ability to modify those weapons. The 56-round _LW-56 magazine_ is the largest 223 magazine which will fit in a standard ammo pouch (`COMPACT_MAG` rather than `BULKY_MAG`).

The only other change is that car seats are somewhat easier to disassemble so as to provide a source of early-game springs.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
> The only other change is that car seats are somewhat easier to disassemble so as to provide a source of early-game springs.

You actually make them harder to disassemble:
Welder type components are replaced just by hacksaw during disassembly.

**Comment by mugling** [comment]
> Welder type components are replaced just by hacksaw during disassembly.

Ah, that I did not know

**Comment by mugling** [comment]
As of fc793ba loaded magazines now have non-trivial weights

**Comment by mugling** [comment]
As of 0cfae12 reloading using a magazine is much quicker.

#14972 applies `item_handling_cost` when fetching magazines with the guns `reload_time` added to this.

Reloading magazines themselves (064304d) is very slow

File: data/json/items/ranged.json [hunk 1]
```diff
@@ -2888,10 +2887,10 @@
     "recoil": 120,
     "durability": 8,
     "burst": 15,
-    "clip_size": 45,
-    "reload": 600,
+    "reload": 100,
     "loudness" : 150,
-    "valid_mod_locations": [[ "accessories", 4 ],[ "barrel", 1 ],[ "bore", 1 ],[ "conversion", 1 ],[ "grip", 1 ],[ "magazine", 2 ],[ "mechanism", 4 ],[ "muzzle", 1 ],[ "rail", 1 ],[ "sights", 1 ],[ "stock", 1 ],[ "underbarrel", 1 ]]
+    "magazines" : [ "stanag50", "stanag30", "stanag10" ],
+    "valid_mod_locations": [[ "accessories", 4 ],[ "barrel", 1 ],[ "bore", 1 ],[ "conversion", 1 ],[ "grip", 1 ],[ "mechanism", 4 ],[ "muzzle", 1 ],[ "rail", 1 ],[ "sights", 1 ],[ "stock", 1 ],[ "underbarrel", 1 ]]
```
Comment (DeadLeaves) [inline data/json/items/ranged.json]: Seeing as this weapon now has a magazine implementation. The description of the weapon should be changed. 
> A H&K416 carbine outfitted with a heavier barrel and an extended magazine...

File: data/json/items/ranged.json [hunk 2]
```diff
@@ -3008,9 +3007,9 @@
     "recoil": 90,
     "durability": 7,
     "burst": 30,
-    "reload": 750,
+    "reload": 400,
     "loudness" : 160,
-    "magazines" : [ "stanag30", "belt223" ],
+    "magazines" : [ "stanag50", "stanag30", "belt223" ],
```
Comment (DeadLeaves) [inline data/json/items/ranged.json]: Assuming this is one of the older variants of m249 that can use STANAG magazines. It should be able to use all STANAG compatible magazines.

File: data/json/items/magazines.json [hunk 3]
```diff
@@ -8,12 +8,62 @@
...
+        "ammo_type" : "223",
+        "capacity" : 10,
+        "reliability" : 9,
+        "reload_time" : 100,
+        "flags" : [ "COMPACT_MAG" ]
+    },
+    {   "type" : "MAGAZINE",
+        "id" : "stanag50",
+        "price" : 9000,
+        "name" : "STANAG drum magazine",
```
Comment (DeadLeaves) [inline data/json/items/magazines.json]: Normally a 50 round STANAG magazine is a normal box magazine. Greater than that we start getting to casket and drum magazines.

Comment (mugling) [inline data/json/items/magazines.json]: This what I had in mind:

m249

The 50-round capacity I want to keep the same so as to maintain overall balance. If you can think of a better name then that can be updated.

Above that I was intending to give the option for ammo belts. Whats a realistic maximum length, I've gone with 200 for now?

Comment (DeadLeaves) [inline data/json/items/magazines.json]: For ammo belts in .223. 200 rounds seems about right.

Regarding the 50 round capacity. I suppose you could name is ``Extended STANAG magazine`` or something. Or make it a casket mag, that still kinda works for 50 rounds.

**Comment by macrosblackd** [comment]
One thing I noticed is that reloading when you have an empty mag in the gun doesn't automatically prompt you to pick a new mag.

**Comment by mugling** [comment]
@macrosblackd on the todo list but not as simple as it sounds

**Comment by macrosblackd** [comment]
Also, I'm getting an issue where if you only have one filled mag left, the game reports that you have no ammo left.

**Comment by mugling** [comment]
@macrosblackd what is the exact sequence of steps and message

**Comment by chaosvolt** [comment]
Hmm. Does this ensure that the guns can potentially spawn with a magazine in them, yet?

**Comment by mugling** [comment]
> Does this ensure that the guns can potentially spawn with a magazine in them, yet?

See the preamble

> Presently they spawn without magazines requiring you to first craft a heavy and unreliable survivor 223 magazine...

**Comment by chaosvolt** [comment]
Ah, doh. x.x

**Comment by mugling** [comment]
This is of course by design as per our earlier discussions. For the 9mm guns which are coming next I'm not so sure.

**Comment by chaosvolt** [comment]
Ah right. For now I've started work on a PR for the weapons I've added (BAR, Sten, and others) so will publish that later on.

**Comment by mugling** [comment]
Superceded by #15072

**Pull request closed without merging**